src/main/java/io/appium/java_client/android/AndroidMobileOptions.java

```diff
@@ -0,0 +1,1733 @@
+     *
+     * @param duration is the number of milliseconds to wait for adb command execution.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ADB_EXEC_TIMEOUT
+     */
+    public AndroidMobileOptions setAdbExecTimeout(Duration duration) {
+        return amend(AndroidMobileCapabilityType.ADB_EXEC_TIMEOUT, duration.toMillis());
+    }
+
+    /**
+     * Get the Timeout in milliseconds used to wait for adb command execution.
+     *
+     * @return Duration to wait for adb command execution.
+     * @see AndroidMobileCapabilityType#ADB_EXEC_TIMEOUT
+     */
+    public Duration getAdbExecTimeout() {
+        Object duration = getCapability(AndroidMobileCapabilityType.ADB_EXEC_TIMEOUT);
+        if (duration.getClass().isAssignableFrom(Long.class)) {
+            return Duration.ofMillis((Long) duration);
+        }
+        Integer integer = (Integer) duration;
+        Long value = Long.valueOf(integer);
+        return Duration.ofMillis(value);
+    }
+
+    /**
+     * Set the port used to connect to the ADB server.
+     *
+     * @param port is the port to connect to the ADB server.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ADB_PORT
+     */
+    public AndroidMobileOptions setAdbPort(Integer port) {
+        return amend(AndroidMobileCapabilityType.ADB_PORT, port);
+    }
+
+    /**
+     * Get the port used to connect to the ADB server.
+     *
+     * @return Integer of the port used to connect to the ADB server.
+     * @see AndroidMobileCapabilityType#ADB_PORT
+     */
+    public Integer getAdbPort() {
+        return (Integer) getCapability(AndroidMobileCapabilityType.ADB_PORT);
+    }
+
+    /**
+     * Set the app to allow installation of a test package which has android:testOnly="true" in the manifest.
+     *
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ALLOW_TEST_PACKAGES
+     */
+    public AndroidMobileOptions setAllowTestPackages() {
+        return setAllowTestPackages(true);
+    }
+
+    /**
+     * Set whether to allow installation of a test package which has android:testOnly="true" in the manifest.
+     *
+     * @param bool when true will be able install a test package marked testOnly.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ALLOW_TEST_PACKAGES
+     */
+    public AndroidMobileOptions setAllowTestPackages(boolean bool) {
+        return amend(AndroidMobileCapabilityType.ALLOW_TEST_PACKAGES, bool);
+    }
+
+    /**
+     * Get whether installation of a test package which has android:testOnly="true" in the manifest is allowed.
+     *
+     * @return true if able to install a test package marked testOnly.
+     * @see AndroidMobileCapabilityType#ALLOW_TEST_PACKAGES
+     */
+    public boolean doesAllowTestPackages() {
+        return (boolean) getCapability(AndroidMobileCapabilityType.ALLOW_TEST_PACKAGES);
```

Comment:
Above we return `Integer` (object type) for a number value, but here we do `boolean` (primitive type). I assume we should follow the common pattern everywhere and return either all primitive (in such case NPE will be thrown if the corresponding value is null) or object types